src/teleop/gui/src/components/CameraSelection.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+    <div class="wrap">
+      <div class="buttons" v-for="j in 3">
+        <template v-for="i in 3">
+          <button class="cam_buttons" v-bind:style="{opacity: opacities[(i-1) + 3*(j-1)]}" v-on:click="$emit('cam_index', (i-1) + 3*(j-1))" :disabled="maxedOut && !camsEnabled[(i-1) + 3*(j-1)]"> <span>{{names[(i-1) + 3*(j-1)]}}</span> </button>
+          <div class="fixed-spacer"></div>
+        </template>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  
+  export default {
+    data() {
+      return {
+        opacities: new Array(9).fill(1.0)
+      }
+    },
+
+    props: {
+      capacity: {
+        type: Number,
+        required: true
+      },
+      camsEnabled: {
+        type: Array,
+        required: true
+      },
+      names: {
+        type: Array,
+        required: true
+      }
+    },
+    
+    computed: {
+      maxedOut() {
+        let num_enabled = 0
+        for (let i = 0; i < this.camsEnabled.length; i++) {
+          if(this.camsEnabled[i]) {
+            num_enabled++;
+          }
+        }
+        return num_enabled == this.capacity;
+      }
+    },
+
+    watch: {
+      camsEnabled() {
```

Comment:
Might be a little less confusing if you style this like in CameraInfo.vue: `camsEnabled: function() { ...`

Then no one thinks camsEnabled is a function